### PR TITLE
Adds a new passive changeling ability - the Defibrillator Grasp. Adds support for 0 cost changeling abilities.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

image

- Adds a new PASSIVE changeling ability, totally not stolen from The Thing, called Defibrillator Grasp. If someone attempts to defibrillate your body while you are in stasis or dead, you will instantly be revived at full strength. Additionally, the person doing the defibbing will have both of their arms removed (or if they're a cyborg they'll just be stunned). 

- Adds support for 0 cost changeling abilities. DNA sting is now 0 cost / optionally learned. 

## Why It's Good For The Game

The intent of this ability to add a new avenue for stealthier changelings to spring upon their victims - faking their death only to get brought to medical, then springing up like a bat out of hell and pouncing upon a now vulnerable medical doctor.

Or maybe pretending to be another type of criminal to get brought to the brig medbay and revived, then suddenly jumping up and striking out.

As for the DNA sting change - Ling gets a ton of buttons to start and people don't often use a lot of them. DNA sting is a prime example. By moving some to just "zero cost, optional", trims down on some action ability bloat. 

## Changelog

:cl: Melbert
add: Added a new 0 cost passive changeling ability, the Defibrillator Grasp. 
add: DNA sing is now no longer innate, but 0 cost, allowing changelings to not take it if they don't plan on using it. 
/:cl:

### CODE REVIEW

**Comment by Jackal-boop** [comment]
is this toggleable?

**Comment by MrMelbert** [comment]
> is this toggleable?

it's a passive

**Comment by Jackal-boop** [comment]
> > is this toggleable?
> 
> it's a passive

doesnt this mean that sec can ID changlings by just killing someone and reviving them

**Comment by MrMelbert** [comment]
> doesnt this mean that sec can ID changlings by just killing someone and reviving them

considering it's optional, IE not all changelings will have it, no, I don't think this will happen

**Comment by chemistrymain2** [comment]
> > doesnt this mean that sec can ID changlings by just killing someone and reviving them
> 
> considering it's optional, IE not all changelings will have it, no, I don't think this will happen

should still be a toggleable even if it's a passive so that you can't get ling tested w it and be fucked just because you bought it

**Comment by githubuser4141** [comment]
> > > doesnt this mean that sec can ID changlings by just killing someone and reviving them
> > 
> > 
> > considering it's optional, IE not all changelings will have it, no, I don't think this will happen
> 
> should still be a toggleable even if it's a passive so that you can't get ling tested w it and be fucked just because you bought it

yeah and you might not want it active in all situations, this is definitely a 'time and place' ability

**Comment by MrMelbert** [comment]
Didn't want it to be a toggle, the idea behind it was requiring some form of commitment. 
Someone suggested allowing a free refund of it which I'm more keen on.

**Comment by ATH1909** [comment]
dna seems really important for lings, not sure if it's a good idea to hide it behind a (0 cost) purchase

**Comment by MrMelbert** [comment]
> dna seems really important for lings, not sure if it's a good idea to hide it behind a (0 cost) purchase

95% of lings either don't use DNA sting at all or use it wrong (sting everyone in sight)

**Comment by DATA-xPUNGED** [comment]
> > dna seems really important for lings, not sure if it's a good idea to hide it behind a (0 cost) purchase
> 
> 95% of lings either don't use DNA sting at all or use it wrong (sting everyone in sight)

How do i use it right, then?

**Comment by AviiCinders** [comment]
> > > is this toggleable?
> > 
> > 
> > it's a passive
> 
> doesnt this mean that sec can ID changlings by just killing someone and reviving them

I think that becomes a policy issue afterwards

**Comment by blackdav123** [comment]
> > > dna seems really important for lings, not sure if it's a good idea to hide it behind a (0 cost) purchase
> > 
> > 
> > 95% of lings either don't use DNA sting at all or use it wrong (sting everyone in sight)
> 
> How do i use it right, then?

sting only useful disguises so that you dont end up getting to many and losing your original body's identity

**Comment by BlueMemesauce** [comment]
> > dna seems really important for lings, not sure if it's a good idea to hide it behind a (0 cost) purchase
> 
> 95% of lings either don't use DNA sting at all or use it wrong (sting everyone in sight)

I don't feel that this is a good idea, since DNA sting is like the whole point of changeling. The reason why most people don't use it is that it's difficult to pull of well. Why would we want to further discourage people from using it? The more people actually decide to use the ability the role is based around, the better.

**Comment by jughu** [comment]
> > > dna seems really important for lings, not sure if it's a good idea to hide it behind a (0 cost) purchase
> > 
> > 
> > 95% of lings either don't use DNA sting at all or use it wrong (sting everyone in sight)
> 
> I don't feel that this is a good idea, since DNA sting is like the whole point of changeling. The reason why most people don't use it is that it's difficult to pull of well. Why would we want to further discourage people from using it? The more people actually decide to use the ability the role is based around, the better.

Because its button bloat the role is orginally intended to be around dna absorptions to complete their objectives but you can complete it with dna stings if anything this encourages lings to absorb more, lings in the past had the upside of getting traitor uplinks etc when you absorbed somoene but since the memory system rework this is not the case anymore leaving dna absorption absolute useless

**Comment by SomeoneYouProbablyKnow** [comment]
Isn't this extra incentive for medical to be even slower when reviving people and just not reviving suspicious bodies with belt defibs? Just stick to metaprotected sec as cmo or get your armes chopped

**Comment by MrMelbert** [comment]
> Isn't this extra incentive for medical to be even slower when reviving people and just not reviving suspicious bodies with belt defibs? Just stick to metaprotected sec as cmo or get your armes chopped

I cannot imagine there is a reality that an optional purchase for a rarer antagonists that does not often come in dead to medical results in medical working slower. 
I highly encourage people to look at the broader picture

**Comment by jughu** [comment]
I think this would really shine if you could re-adapt or pick you abilities while your dead you can only re-adapt with dna absorptions and will give dna absorbtions a better reason to be use

**Comment by CCC23** [comment]
I love this

**Comment by moocowswag** [comment]
> > dna seems really important for lings, not sure if it's a good idea to hide it behind a (0 cost) purchase
> 
> 95% of lings either don't use DNA sting at all or use it wrong (sting everyone in sight)

Its not wrong if it causes half the station to mald / rq while ai no longer recognizes them as human.

**Comment by MrMelbert** [comment]
> Its not wrong if it causes half the station to mald / rq while ai no longer recognizes them as human.
> 
> dna sting is the single most straightforward way to impact a round as a ling you dont need it for stealth or trickery
> 
> youre basically a weaker version of a wizard running around with polymorph staff

Wrong sting. You're talking about transformation sting.

File: code/modules/antagonists/changeling/powers/fakedeath.dm [hunk 1]
```diff
@@ -3,36 +3,76 @@
...
 	button_icon_state = "fake_death"
 	chemical_cost = 15
-	dna_cost = 0
+	dna_cost = CHANGELING_POWER_INNATE
 	req_dna = 1
 	req_stat = DEAD
 	ignores_fakedeath = TRUE
+
+	var/fakedeath_duration = 40 SECONDS
 	var/revive_ready = FALSE
```
Comment (Jacquerel) [inline code/modules/antagonists/changeling/powers/fakedeath.dm:12]: Be a dear and document these variables

File: code/modules/antagonists/changeling/powers/defib_grasp.dm [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	changeling.emote("scream")
+	playsound(changeling, 'sound/magic/demon_consume.ogg', 50, TRUE)
+
+	// Mimics some real defib stuff (wish this was more generalized)
+	playsound(defib, SFX_BODYFALL, 50, TRUE)
+	playsound(defib, 'sound/machines/defib_zap.ogg', 75, TRUE, -1)
+	playsound(defib, 'sound/machines/defib_success.ogg', 50, FALSE) // I guess
+	defib.shock_pulling(30, changeling)
+
+/datum/action/changeling/defib_grasp/proc/remove_arms(mob/living/carbon/changeling, mob/living/defibber, obj/item/shockpaddles/defib)
```
Comment (Jacquerel) [inline code/modules/antagonists/changeling/powers/defib_grasp.dm:47]: These procs too

**Review by Jacquerel — CHANGES_REQUESTED** [review]
Great add, don't have anything to mention except to be picky

**Pull request merged**